hosting/src/hooks/useTanamDocuments.tsx

```diff
@@ -111,6 +122,43 @@ export function useCrudTanamDocument(documentId?: string) {
       setIsLoading(false);
     }
   }
-
   return {update, isLoading, error};
 }
+
+export function useCreateTanamDocument(documentType?: string) {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<UserNotification | null>(null);
+
+  async function create() {
+    setIsLoading(true);
+    try {
+      if (!documentType) {
+        setError(new UserNotification("error", "Missing parameter", "Document id parameter is missing"));
+        return;
+      }
+      const typeRef = collection(firestore, "tanam-documents");
+      const result = await addDoc(typeRef, {
+        createdAt: serverTimestamp(),
+        data: {content: ""},
+        documentType,
+        publishedAt: null,
+        revision: 0,
+        status: "",
+        updatedAt: serverTimestamp(),
+      });
```

Comment:
Create a new instance of the class and run "toJson" on it. That was the reason why you made `createdAt` nullable.